bindings/nodejs/src/lib.rs

```diff
@@ -50,66 +78,189 @@ pub struct ObjectMeta {
+}
+
+#[napi]
+impl ObjectLister {
+    pub fn new(op: opendal::ObjectLister) -> Self {
         Self { inner: op }
     }
 
     #[napi]
-    pub async fn meta(&self, path: String) -> Result<ObjectMeta> {
-        let o = self.inner.object(&path);
-        let meta = o
+    pub async unsafe fn next_page(&mut self) -> Result<Vec<DataObject>> {
+        Ok(self.inner
+            .next_page()
+            .await
+            .map_err(format_napi_error)?.unwrap_or_default()
+            .iter()
+            .map(|obj| DataObject { inner: obj.to_owned()})
+            .collect())
+    }
+}
+
+#[napi]
+pub struct DataObject {
+    inner: opendal::Object
+}
+
+#[napi]
+impl DataObject {
+    pub fn new(op: opendal::Object) -> Self {
+        Self { inner: op }
+    }
+
+    #[napi]
+    pub async fn stat(&self) -> Result<ObjectMetadata> {
+        let meta = self.inner
             .stat()
             .await
             .map_err(format_napi_error)
             .unwrap();
 
-        let (secs, nsecs) = meta
-            .last_modified()
-            .map(|v| (v.unix_timestamp(), v.nanosecond()))
-            .unwrap_or((0, 0));
-
-        Ok(ObjectMeta {
-            location: path,
-            last_modified: DateTime::<Utc>::from_utc(
-                NaiveDateTime::from_timestamp_opt(secs, nsecs)
-                    .expect("returning timestamp must be valid"),
-                Utc,
-            ).timestamp(),
-            size: meta.content_length() as u32
-        })
+        exact_meta(meta)
     }
 
-    #[napi]
-    pub async fn write(&self,  path: String, content: Vec<u8>) -> Result<()> {
-        self.inner.object(&path)
-            .write(content)
+    #[napi(js_name="statSync")]
+    pub fn blocking_stat(&self) -> Result<ObjectMetadata> {
+        let meta = self.inner
+            .blocking_stat()
             .map_err(format_napi_error)
+            .unwrap();
+
+        exact_meta(meta)
+    }
+
+    #[napi]
+    pub async fn write(&self, content: Buffer) -> Result<()> {
+        let c = content.as_ref().to_owned();
+        self.inner
+            .write(c)
             .await
+            .map_err(format_napi_error)
+    }
+
+    #[napi(js_name="writeSync")]
+    pub fn blocking_write(&self, content: Buffer) -> Result<()> {
+        let c = content.as_ref().to_owned();
+        self.inner.blocking_write(c).map_err(format_napi_error)
     }
 
     #[napi]
-    pub async fn read(&self,  path: String) -> Result<Vec<u8>> {
-        let res = self.inner.object(&path)
+    pub async fn read(&self) -> Result<Buffer> {
+        let res = self.inner
             .read()
             .await
             .map_err(format_napi_error)
             .unwrap();
-        Ok(res)
+        Ok(res.into())
+    }
+
+    #[napi(js_name="readSync")]
+    #[napi]
+    pub fn blocking_read(&self) -> Result<Buffer> {
+        let res = self.inner
+            .blocking_read()
+            .map_err(format_napi_error)
+            .unwrap();
```

Comment:
How about use `?` instead of `unwrap()`?

Reply:
Great!